sql/2020/04_Fonts/04_07.font_resource_hints_with_fcp.sql

```diff
@@ -0,0 +1,88 @@
+#standardSQL
+#font_resource_hints_with_fcp(??NoResult)
+CREATE TEMPORARY FUNCTION getResourceHints(payload STRING)
+RETURNS ARRAY < STRUCT < name STRING, href STRING >>
+    LANGUAGE js AS '''
+var hints = new Set(['preload', 'prefetch', 'preconnect', 'prerender', 'dns-prefetch']);
+try {
+    var $ = JSON.parse(payload);
+    var almanac = JSON.parse($._almanac);
+    return almanac['link-nodes'].reduce((results, link) => {
```

Comment:
The query is returning 0 results because this line is calling `reduce` on something that isn't an array, so the `catch` handler below returns an empty array.

```suggestion
    return almanac['link-nodes'].nodes.reduce((results, link) => {
```

Reply:
Thanks for the suggestion which is working well,
however, the things seem to go a wrong way again being return 0 when I try to join CrUX report.
May I have your help in direction?

Reply:
@AbbyTsai I ran the latest version of the query and it produces [results](https://docs.google.com/spreadsheets/d/1OsAogFyPFOkYF185pHHV42mQ87XFCmU5LE3PUnDHlFQ/edit?usp=sharing). There are a few things to clean up or optimize, but it seems to be working.